### DISCUSSION TITLE
Can Uno build and deploy to iOS 10 devices (e.g. iPad 4th gen)?

### DISCUSSION THREAD

**Discussion by jeromelaban** [unoplatform/uno#9186] [Q&A] [answered]
[_From Discord_](https://discord.com/channels/372137812037730304/550416151172087808/992040832066670706)

Hi all, Can Uno build and deploy to iOS 10 devices (e.g. iPad 4th gen)? If so how do I configure things so it can build and deploy to one? Currently I'm getting an "invalid architecture" error with ARM64 and "Uno.UI.dll" apparently cannot build for ArmV7 or ArmV7s.

**Comment by jeromelaban** [comment, marked as answer]
Uno is based on either Xamarin or .NET 6, and supports what both of those technologies support. You may still need to build with latest SDKs in most cases, but you can specify a min supported version in you Info.plist.

Note that in some cases, features may be available in earlier iOS versions (as they did not exist and cannot be emulated by Uno).

**Reply by Glacial** [reply]
Ok, so what do I do? My project is currently Xamarin style I think (from the template with multiple project heads) but how can I tell? An iPad (4th gen) is apparently ARMv7/ARMv7s architecture but attempting to build with ARMv7 + ARM64 set results in the following error:

Xamarin.iOS.Common.targets(235,3): error MT5107: The assembly 'Uno.UI.dll' can't be AOT-compiled for 32-bit architectures because the native code is too big for the 32-bit ARM architecture.

There's a bunch of "error GDC116A36: can not encode offset '0x10056A4' in resulting scattered relocation." errors too.

**Reply by jeromelaban** [reply]
Ah, I see, the larger error message is important indeed. The only way to fix this is to ensure that you have the IL Linker running to trim the assemblies down to a size that the tooling will accept.

Otherwise, if the IL Linking does not help, we have no solution for this other than moving to ARM64, as ARMv7 devices have been out of support by Apple for a long while.

**Reply by Glacial** [reply]
Thanks. I gave full linking a try and the same issues were seen. 
I suspected devices this old wouldn't be supported but perhaps the official line of "runs on iOS 8 and above" on the requirements page needs updating.

**Reply by jeromelaban** [reply]
That's fair. We'll update the docs, thanks for the feedback.
